### ISSUE TITLE
Strange Mass Control slider behaviour

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab#187]
While verifying https://github.com/phetsims/gravity-force-lab/issues/155 I noticed some strange behaviour on the mass control sliders when using regular keyboard shortcuts. This might be a Mac OS issue - not sure.

Mac OS 10.13.6

Issues using Home and End keys on **Mass 1** (fn + left arrow or fn +right arrow)
- Using Home to jump to minimum I heard, "Mass 1" instead of "10 kilograms"
- Using End to jump to maximum I sometimes heard, "1000 kilograms, Mass 1, slider", and sometimes hear, "10 kilograms, Mass 1, slider".

Issues using Home and End keys on **Mass 2** (fn + left arrow or fn +right arrow)
- Using Home to jump to minimum I heard, "Mass 1" instead of "10 kilograms"
- Using End to jump to maximum I sometimes heard, "1000 kilograms, Mass 2, slider", and never heard, "10 kilograms, Mass 2, slider".

I don't know why I am hearing "Mass 1" when using Home key.
I don't know why I am sometimes hearing 10 kilograms instead 1000.
I don't know why I am getting the label and function after focus.

@zepumph, any thoughts?

**Comment by zepumph** [comment]
This is very weird! I don't get any of these oddities on windows 10 chrome/firefox nvda. When pressing home/end, I consistently get "1000 kilograms." And I only get the (correct) label on focus (Mass 1 slider 1000 kilograms).

Are you able to consistently reproduce this?

**Comment by terracoda** [comment]
It happens every time I use the Home and End key shortcuts.

**When focus is on Mass 1 slider:**
- Pressing Home key -> I get "Mass 1", then right after a context response (alert) with the correct force value. 
- Pressing End key -> I get "10 kilograms, Mass 2, slider" (as if I moved focus to Mass 2, though I haven't), then again right after I get the correct context response with the correct force value.

**When focus is on Mass 2 slider:**
- Pressing Home key -> I get "Mass 1", then right after a context response (alert) with the correct force value. 
- Pressing End key -> I get "1000 kilograms, Mass 2, slider" (as if I moved focus to Mass 2, thought I haven't), then again right after I get the correct context response with the correct increased force value.

Upon focus of both controls, I get the correct object responses with the correct name and value for each slider.

Could this have anything to do with the over-reaching bounds mentioned over in #194?

**Comment by terracoda** [comment]
And, one time, I managed to  grabbed a screen shot.
Note where the pink focus highlight is, and what VO is saying. I had just pressed the End key.
<img width="814" alt="Screen Shot 2019-11-11 at 10 41 22 PM" src="https://user-images.githubusercontent.com/1902116/68638387-f69c2a80-04db-11ea-80d3-65512dc95556.png">

**Comment by zepumph** [comment]
@jessegreenberg and I were able to crack this one today. The issue is only with home/end keys because VO seems to do something with those shortcuts. In shortcuts doc, it says that `VO + Home` goes to the top of an area, and end goes to bottom of it. It seems like there is a VO bug where the `fn + left/right arrow` keys are somehow being mapped into the `VO+home`. This is buggy in a few ways.

* The most obvious is if you have the VO Highlight showing (which I would recommend @terracoda doing and reproducing the bug). The VO Highlight moves away from the slider in use, even though the dom focus doesn't (very weird!).
* This seems like a bug with VO in which we should have control of the home/end keys, but they are being "half" co-opted, in which it still moves the slider, AND it triggers VO key stroke behavior. At the very least I would like it to do the VO behavior, and then not pass the home/end keystroke to the sim (do one or the other, not both). I will submit a bug report to apple for this.

**Comment by zepumph** [comment]
It would be nice to discuss this during our meeting @terracoda.

**Comment by zepumph** [comment]
@terracoda and I spoke about this more today. We were very curious about what made this an issue for the mass controls, but not the position sliders. After some investigation, we found that having a label on the div around the mass controls was turning it into a "group" that was triggering the VO bug where home/end would go to the front/end of the group. Getting rid of the group, fixed this bug for us. 
We should still consider this a workaround though, for our case. I think that @terracoda would have preferred to have the group label on it, though having non-buggy interaction is most important. I'll still submit a bug to apple.

**Comment by zepumph** [comment]
@terracoda can you confirm this is fixed in master. If all is good with this particular case, then we can close.

**Comment by terracoda** [comment]
Without the group identification (aria-labeledby assocaition with the h3 Mass Controls) the sliders are working fine with Home and End keys. Focus changes are working correctly.

And VO keys plus fn-left/right now do their VO thing going to start and end of sim's main article element.

Closing issue. Thanks @zepumph!
